robotbase/src/main/java/frc/robot/subsystems/Laterator.java

```diff
@@ -0,0 +1,145 @@
+package frc.robot.subsystems;
+
+import static edu.wpi.first.units.Units.Inches;
+import static edu.wpi.first.units.Units.Millisecond;
+import static edu.wpi.first.units.Units.Percent;
+import static edu.wpi.first.units.Units.Rotations;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.Dimensionless;
+import edu.wpi.first.units.measure.Distance;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.LATERATOR;
+
+public class Laterator extends SubsystemBase {
+
+  private SparkMax m_motor;
+
+  private ProfiledPIDController m_PIDController;
+  private RelativeEncoder m_encoder;
+
+  private DigitalInput m_hallEffectSensor;
+  private Debouncer m_debouncer;
+  private boolean m_isHallEffectTriggered;
+
+  private Angle m_targetRotations = Rotations.of(Double.NaN);
+
+  public Laterator() {
+    m_motor = new SparkMax(
+      Constants.CAN_ID.LATERATOR_MOTOR,
+      MotorType.kBrushless
+    );
+    m_motor.configure(
+      Constants.LATERATOR.MOTOR_CONFIG,
+      ResetMode.kResetSafeParameters,
+      PersistMode.kPersistParameters
+    );
+
+    m_PIDController = new ProfiledPIDController(
+      LATERATOR.MOTOR_P,
+      LATERATOR.MOTOR_I,
+      LATERATOR.MOTOR_D,
+      new TrapezoidProfile.Constraints(LATERATOR.MAX_VEL, LATERATOR.MAX_ACCEL)
+    );
+
+    m_encoder = m_motor.getEncoder();
+
+    m_hallEffectSensor = new DigitalInput(
+      Constants.DIGITAL_INPUT.LATERATOR_HALL_EFFECT_ID
+    );
+    m_debouncer = new Debouncer(
+      Constants.LATERATOR.DEBOUNCE_TIME_SECONDS.in(Millisecond)
+    );
+  }
+
+  public void setSpeed(Dimensionless speed) {
+    m_targetRotations = Rotations.of(Double.NaN);
```

Comment:
I believe we want to be using mutable units in this case for performance reasons: https://docs.wpilib.org/en/stable/docs/software/basic-programming/java-units.html#mutability-and-object-creation